core/src/main/java/org/commonjava/maven/galley/internal/xfer/DownloadHandler.java

```diff
@@ -238,6 +229,14 @@ else if ( !suppressFailures )
         finally
         {
             transferSizes.remove( target );
+            if ( created )
+            {
+                synchronized ( DOWNLOAD_MUTEX )
+                {
+                    logger.debug( "Removing download job of path: {}: {}", resource, future );
+                    pending.remove( target );
+                }
+            }
```

Comment:
I think the fix can work but it will bring some side effect which will make download conflict. Because there is no retry check for that pending remove, so when a big file download timeout, it will trigger a retry with current future. But pending has been removed now so if there is a new download coming in the meantime, a new future will be created to do downloading for same resource, which could bring download conflict. (or maybe a not completed target will be returned)
Maybe we need to check tries>0 before the code.

Reply:
Double checked and seems that retry is happening in inner loop. So ignore this.